processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -74,7 +88,12 @@ func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schem
 	d.rb.SetCloudProvider(conventions.AttributeCloudProviderAWS)
 	d.rb.SetCloudPlatform(conventions.AttributeCloudPlatformAWSEKS)
 
-	return d.rb.Emit(), conventions.SchemaURL, nil
+	if d.ra.K8sClusterName.Enabled {
+		clusterName := d.utils.getClusterName(ctx, d.logger)
+		d.rb.SetK8sClusterName(clusterName)
+	}
+
+	return d.rb.Emit(), conventions.SchemaURL, err
```

Comment:
```suggestion
	return d.rb.Emit(), conventions.SchemaURL, nil
```

The only reason I am suggesting this is that the `isEKS()` doesn't actually handle the error directly. I'm not sure if it's possible but there could be a scenario where `isEKS == true` and `err != nil`. In this case you would be returning an error here and the resources would be dropped by caller. 

I think we could return nil here and/or check for non nil error above.

Reply:
Fair point. There currently isn't any scenario where `err` would be non-nil, and `isEKS` would be true, but I could see the confusion.